include/cuco/detail/__config

```diff
@@ -32,3 +46,7 @@
 #if (CUCO_CUDA_MINIMUM_ARCH >= 700)
 #define CUCO_HAS_INDEPENDENT_THREADS
 #endif
+
+#if defined(__SIZEOF_INT128__)
```

Comment:
Both gcc and clang support 128-bit integers, the check seems redundant.

Reply:
I agree the check seems a bit overkill. However, without the check, the error messages you'll get if a target arch/compiler doesn't support 128bit ints are very weird:
When I compile my `fast_int` implementation with nvcc 11.4 (which doesn't support int128) I get
```
<source>(80): warning: missing return statement at end of non-void function "util::fast_int<T>::log2 [with T=int64_t]"
          detected during:
            instantiation of "util::fast_int<T>::value_type util::fast_int<T>::log2(util::fast_int<T>::value_type) const noexcept [with T=int64_t]" 
(85): here
            instantiation of "void util::fast_int<T>::evaluate_magic_numbers() noexcept [with T=int64_t]" 
(28): here
            instantiation of "util::fast_int<T>::fast_int(T) noexcept [with T=int64_t]" 
(150): here
```
which isn't really helpful for finding out where the error comes from.